src/main/java/com/akukhtin/springbootapplication/service/impl/UserServiceImpl.java

```diff
@@ -12,8 +14,13 @@
     private UserRepository userRepository;
 
     @Override
-    public User update(User user) {
-        return userRepository.save(user);
+    public Optional<User> update(Long id, UserDto userDto) {
+        User newUser = userRepository.findById(id).orElseThrow();
+        newUser.setName(userDto.getName());
+        newUser.setSurname(userDto.getSurname());
+        newUser.setPassword(userDto.getPassword());
+        newUser.setCountry(userDto.getCountry());
+        return Optional.of(newUser);
```

Comment:
@AlexKukhtinDev ти ж тут впевнений в тому що user != null.... тоді навіщо ти обгортаєш його в optional?

```suggestion
        return newUser;
```